source/Applications/Advanced/HandEyeCalibration/UtilizeEyeInHandCalibration/UtilizeEyeInHandCalibration.cpp

```diff
@@ -97,10 +127,45 @@ int main()
         std::cout << "Computing camera pose in robot base frame" << std::endl;
         const Eigen::MatrixXf transformBaseToCamera = transformBaseToEndEffector * transformEndEffectorToCamera;
 
-        std::cout << "Computing (picking) point in robot base frame" << std::endl;
-        const Eigen::Vector4f pointInBaseFrame = transformBaseToCamera * pointInCameraFrame;
-        std::cout << "Point coordinates in robot base frame: " << pointInBaseFrame.x() << " " << pointInBaseFrame.y()
-                  << " " << pointInBaseFrame.z() << std::endl;
+        switch(enterCommand())
+        {
+            case CommandType::cmdTransformSinglePoint:
+            {
+                // define (picking) point in camera frame
+                const Eigen::Vector4f pointInCameraFrame(81.2F, 18.0F, 594.6F, 1.0F);
+                std::cout << "Point coordinates in camera frame: " << pointInCameraFrame.x() << " "
+                          << pointInCameraFrame.y() << " " << pointInCameraFrame.z() << std::endl;
+
+                std::cout << "Computing (picking) point in robot base frame" << std::endl;
+                const Eigen::Vector4f pointInBaseFrame = transformBaseToCamera * pointInCameraFrame;
+                std::cout << "Point coordinates in robot base frame: " << pointInBaseFrame.x() << " "
+                          << pointInBaseFrame.y() << " " << pointInBaseFrame.z() << std::endl;
+                break;
+            }
+            case CommandType::cmdTransformPointCloud:
+            {
+                const auto dataFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/ZividGem.zdf";
+                std::cout << "Reading ZDF frame from file: " << dataFile << std::endl;
+                const Zivid::Frame frame = Zivid::Frame(dataFile);
+
+                std::cout << "Getting point cloud from frame" << std::endl;
+                auto pointCloud = frame.pointCloud();
```

Comment:
I think this section should be done before line 130 -> switch(enterCommand()). Then, instead of having hardcoded 3D coordinate points in the line 135, we should have hardcoded image coordinates. The 3D coordinates should be extracted from the point cloud based on the hardcoded image coordinates. The image coordinates should be (1357 666) if I am correct that should be the centroid of the zivid gem.

Reply:
Done. Hardcoded image coordinates and extracts the x,y,z values.